docker/pandoc-lambda/function/app.py

```diff
@@ -123,6 +127,166 @@ def promote_case_footnotes(doc):
 
 
 
+def sectionizer(doc, doc_w_twips=10080, doc_h_twips=14400, paracols=2, internal_margin_twips=720, external_margin_twips=1080, big_margin_twips=1440):
+    """
+        Adds section breaks for headers, footers, columns, etc.
+
+        doc_w doc_h:
+            document width and height in TWIPS
+            1 TWIP = 1/20th pt. (a typographical point)
+            1 inch = 1440 TWIPS
+            my local grocer is about 1300 ft. in each direction, so 1 grocery trip ~= 6.5e+7 grocery TWIPs
+
+        paracols:
+            Number of columns in non-header paragraph text. There is no way to specify # of cols in a style
+    """
+
+    doc_w = Twips(doc_w_twips)
+    doc_h = Twips(doc_h_twips)
+    internal_margin = Twips(internal_margin_twips)
+    external_margin = Twips(external_margin_twips)
+    big_margin = Twips(big_margin_twips)
+
+    # See which rIDs pandoc gave our headers and footers. These values come
+    rels = doc.part.rels
+    headers_and_footers = { rels[rid].target_ref.replace('.xml', '') : rels[rid]
+      for rid in rels if rels[rid].target_ref.endswith('.xml') and
+      # rels lists, among other things, related xml files like header and footer.
+      # target_ref property holds the file name. Check the docx word/_rels/document
+      (rels[rid].target_ref.startswith('header') or rels[rid].target_ref.startswith('footer'))
+    }
+
+    body_element = doc.element.xpath('/w:document/w:body')[0]
+
+    def section_break(destination, frontmatter=False, chapter=False, section=False, paragraph=False):
+        """ Assembles the section break, appends it to the appropriate graf, properly discards the empty wrapper,
+        because it gives a hoot and won't pollute. """
+
+        # The doc itself should have an existing default section attached to body, not a paragraph
+        if destination == 'doc-wide':
+            sec = doc.sections[0]
+        else:
+            sec = doc.add_section()
+            destination._element.xpath('w:pPr')[0].append(sec._sectPr)
+            body_element.remove(doc.paragraphs[-1]._element)
+
+        if chapter:
+            sec.start_type = WD_SECTION_START.ODD_PAGE
+        elif section:
+            sec.start_type = WD_SECTION_START.NEW_PAGE
+        else:
+            sec.start_type = WD_SECTION_START.CONTINUOUS
+
+        sec.bottom_margin = internal_margin
+        sec.gutter = internal_margin // 2
+        sec.header_distance = external_margin
+        sec.left_margin = external_margin
+        sec.orientation = WD_ORIENTATION.PORTRAIT
+        sec.page_height = doc_h
+        sec.page_width = doc_w
+        sec.right_margin = external_margin
+        sec.top_margin = big_margin
+
+        # don't need any of this stuff for just a block of paragraphs. There's defintiely a better way to differentiate
+        # who does and doesn't need headers
```

Comment:
Can you elaborate what's undesirable there? You say, "There's defintiely a better way"?

Reply:
Whoops— cleaned up the code but not the comment. What I have in place works well.